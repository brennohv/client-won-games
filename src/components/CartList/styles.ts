import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import * as EmptyStyles from 'components/Empty/styles'

type WrapperProps = {
  isEmpty: boolean
}

export const Wrapper = styled.main<WrapperProps>`
  ${({ theme, isEmpty }) => css`
    background: ${theme.colors.white};
    display: flex;
    flex-direction: column;
    align-self: start;

    ${isEmpty &&
    css`
      ${EmptyStyles.Wrapper} {
        padding-bottom: ${theme.spacings.medium};
      }
      ${EmptyStyles.Image} {
        max-width: 20rem;
      }
      ${EmptyStyles.Ttile} {
        font-size: ${theme.font.sizes.large};
      }
      ${EmptyStyles.Description} {
        color: ${theme.colors.black};
        font-size: ${theme.font.sizes.medium};
      }
    `}
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #f3f3f3;
    padding: ${theme.spacings.xsmall};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
      padding: ${theme.spacings.small};
    `}
  `}
`

export const Total = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
  `}
`
export const GameList = styled.div`
  max-height: 40rem;
  overflow-y: auto;
`
