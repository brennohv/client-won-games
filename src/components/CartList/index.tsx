import GameItem, { GameItemProps } from 'components/GameItem'
import Button from 'components/Button'

import Link from 'next/link'

import * as S from './styles'
import Empty from 'components/Empty'

export type CartListProps = {
  gamesCart?: GameItemProps[]
  total?: string
  hasButton?: boolean
}

const CartList = ({
  gamesCart = [],
  total,
  hasButton = false
}: CartListProps) => (
  <S.Wrapper isEmpty={!gamesCart.length}>
    {gamesCart.length ? (
      <>
        {gamesCart.map((gameCart) => (
          <GameItem key={gameCart.title} {...gameCart} />
        ))}

        <S.Footer>
          {!hasButton && <span>Total:</span>}
          <S.Total>{total}</S.Total>
          {hasButton && (
            <Link href="/cart" passHref>
              <Button as="a">Buy it now</Button>
            </Link>
          )}
        </S.Footer>
      </>
    ) : (
      <Empty
        description="Go back to the store and explore games and offers"
        title="You do not have games on cart"
        hasLink
      />
    )}
  </S.Wrapper>
)

export default CartList
