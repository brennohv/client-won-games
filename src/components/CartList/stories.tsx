import { Story, Meta } from '@storybook/react'
import CartList, { CartListProps } from '.'
import gameCart from './mock'

export default {
  title: 'CartList',
  component: CartList,
  args: {
    gamesCart: gameCart,
    total: '330,00'
  },
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  }
} as Meta

export const Default: Story<CartListProps> = (args) => <CartList {...args} />
