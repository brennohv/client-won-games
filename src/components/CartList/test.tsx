import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import CartList from '.'

const props = {
  gamesCart: [
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x70',
      title: 'Red Dead Redemption 2',
      price: 'R$ 215,00'
    },
    {
      img: 'https://source.unsplash.com/user/willianjusten/151x70',
      title: 'Borderlands 3',
      price: 'R$ 215,00'
    }
  ],
  total: 'R$ 330,00'
}

describe('<CartList />', () => {
  it('should render the correctly', () => {
    const { container } = renderWithTheme(<CartList {...props} />)

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText('R$ 330,00')).toHaveStyle({
      color: theme.colors.primary
    })
    expect(screen.getByText('Total:')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
