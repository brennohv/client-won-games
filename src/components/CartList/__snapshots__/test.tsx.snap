// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the cart list 1`] = `
.c2 {
  background-color: #FAFAFA;
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  max-width: 9.4rem;
  max-height: 5.6rem;
  width: 100%;
}

.c5 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
}

.c7 {
  color: #030517;
  font-size: 1.4rem;
  line-height: 1.4rem;
  margin-bottom: 0.8rem;
}

.c9 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 600;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
}

.c8 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f3f3f3;
  padding: 1.6rem;
  font-weight: 600;
  font-size: 1.6rem;
}

.c11 {
  color: #F231A5;
}

.c1 {
  max-height: 40rem;
  overflow-y: auto;
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c4 {
    margin-right: 2.4rem;
    max-width: 15.1rem;
    max-height: 7rem;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 1.6rem;
    line-height: 1.6rem;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 2.0rem;
    padding: 2.4rem;
  }
}

<main
  class="c0"
  data-cy="cart-list"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-cy="game item"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Red Dead Redemption 2"
            class="c5"
            role="img"
            src="https://source.unsplash.com/user/willianjusten/151x70"
          />
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Red Dead Redemption 2
          </h3>
          <div
            class="c8"
          >
            <div
              class="c9"
            >
              R$ 215,00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c2"
      data-cy="game item"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Borderlands 3"
            class="c5"
            role="img"
            src="https://source.unsplash.com/user/willianjusten/151x70"
          />
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Borderlands 3
          </h3>
          <div
            class="c8"
          >
            <div
              class="c9"
            >
              R$ 215,00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <span>
        Total:
      </span>
      <span
        class="c11"
      >
        R$ 330,00
      </span>
    </div>
  </div>
</main>
`
